### PR TITLE
fix(dialog): wrap instead of truncating

### LINKED ISSUE #2333: lightbox dialog: Title should wrap instead of using ellipses
### I verified there's no existing issue for this bug.

- [X] There are no existing issues

### Current behavior

Text in the title of lightbox dialog is truncated with ellipses instead of wrapping

### Expected behavior

Title text should wrap

### Skin version

17.4.1

### Device/OS/Browser info

All devices & browsers

### Steps to reproduce

```bash
Create lightbox dialog with header longer than screen width
```

### Relevant code block

_No response_

### Additional context

Text wrap was the initial behavior for this component, but there was a push from design systems team to truncate with ellipses instead. This should not have been approved because, especially after internationalization, there are likely to be cases where the title is impossible to read for some users.

Even in the worst case with text wrapping, the body will still be visible:

image

https://github.com/eBay/skin/assets/26027232/3ada6517-fda4-492d-928a-3d27859b1e00

As such, title should be reverted back from truncation to wrapping.

### PR DESCRIPTION
**Pull request by LuLaValva**
- Fixes #2333

- [x] This PR contains CSS changes
- [ ] This PR does not contain CSS changes

## Description
- Wrap instead of truncating titles in dialogs.
- Ensure that any icons/buttons on the same flex row as the title align to top instead of center.

## Screenshots
image
image
image
image
image

## Checklist

- [x] I verify the build is in a non-broken state
- [x] I verify all changes are within scope of the linked issue

- [x] I regenerated all CSS files under dist folder
- [ ] I tested the UI in all supported browsers
- [ ] I did a visual regression check of the components impacted by doing a Percy build and approved the build
- [x] I tested the UI in dark mode and RTL mode
- [x] I added/updated/removed Storybook coverage as appropriate

### CODE REVIEW

**Comment by agliga** [comment]
Lets target feature branch. This is more of a feature than a bug.

File: dist/drawer-dialog/drawer-dialog.css [hunk 1]
```diff
@@ -94,11 +91,13 @@
...
-  background-color: transparent;
+button.icon-btn.drawer-dialog__close {
+  align-self: start;
   border: 0;
-  height: auto;
-  outline-offset: -8px;
+  height: 32px;
+  min-width: 32px;
+  position: relative;
+  width: 32px;
```
Comment (agliga) [inline dist/drawer-dialog/drawer-dialog.css:100]: Why was this changed? 
I believe in drawer we allow putting in text possibly in the button, so this might break that. 
Again, this was done before icon button existed so just need to make sure this doesn't break that case.

File: src/less/mixins/private/dialog-mixins.less [hunk 2]
```diff
@@ -71,9 +71,6 @@
         align-self: center;
         flex: 1 1 auto;
         margin: 0;
-        overflow: hidden;
-        text-overflow: ellipsis;
-        white-space: nowrap;
```
Comment (agliga) [inline src/less/mixins/private/dialog-mixins.less:76]: In old iterations we used `line-clamp` https://css-tricks.com/almanac/properties/l/line-clamp/ which we can use to limit it to 2-3 lines.

**Review by agliga — COMMENTED** [review]
LGTM overall, couple of comments

File: src/less/mixins/private/dialog-mixins.less (hunk 2 above)
Comment (ArtBlue) [reply src/less/mixins/private/dialog-mixins.less:76]: I agree with clamping, but we had a long discussion about this offline (see thread) and it was decided to wrap indefinitely.

File: src/less/drawer-dialog/drawer-dialog.less [hunk 3]
```diff
@@ -66,11 +66,13 @@
 
 // inherits from .icon-btn
 // Added icon button selector so it doesn't override fake-link
-button.icon-button.drawer-dialog__close {
-    background-color: transparent;
+button.icon-btn.drawer-dialog__close {
+    align-self: start;
```
Comment (ArtBlue) [inline src/less/drawer-dialog/drawer-dialog.less:70]: Why is the `background-color: transparent` being removed? Seems out of scope.

File: src/less/drawer-dialog/drawer-dialog.less [hunk 4]
```diff
@@ -66,11 +66,13 @@
...
     border: 0;
-    height: auto;
-    outline-offset: -8px;
+    height: 32px;
```
Comment (ArtBlue) [inline src/less/drawer-dialog/drawer-dialog.less:72]: When I initially tried out the change statically in dev tools, all that was needed was the `align-self: start`. Maybe it's relevant for other use cases, but unless it is (we should make sure), we should avoid creating restrictions.

File: src/less/toast-dialog/toast-dialog.less [hunk 5]
```diff
@@ -65,8 +65,9 @@
 
 // inherits from .icon-btn
 button.toast-dialog__close {
-    align-self: center;
+    align-self: start;
     border: 0;
+    flex-shrink: 0;
```
Comment (ArtBlue) [inline src/less/toast-dialog/toast-dialog.less:70]: Is this needed? What are the implications if it's not included?

**Review by ArtBlue — COMMENTED** [review]
Can't test locally yet (build issue), but some thoughts...

**Review by agliga — CHANGES_REQUESTED** [review]
Also found two bugs we need to fix

Drawer dialog close button text does not float to top
Screenshot 2024-05-14 at 9 46 32 AM

Panel dialog with close text overflows on the x axis
Screenshot 2024-05-14 at 9 47 33 AM
Screenshot 2024-05-14 at 9 47 37 AM

File: dist/drawer-dialog/drawer-dialog.css (hunk 1 above)
Comment (LuLaValva) [reply dist/drawer-dialog/drawer-dialog.css:100]: So this actually wasn't doing anything at all before, since `icon-button` should have been `icon-btn`. For these changes I pulled in all of our updates from lightbox dialog.

File: src/less/drawer-dialog/drawer-dialog.less (hunk 3 above)
Comment (LuLaValva) [reply src/less/drawer-dialog/drawer-dialog.less:70]: Same as above, this selector was previously doing nothing since `icon-button` should actually be `icon-btn` so I pulled in current `lightbox-dialog` styles.

File: src/less/drawer-dialog/drawer-dialog.less (hunk 4 above)
Comment (LuLaValva) [reply src/less/drawer-dialog/drawer-dialog.less:72]: Also changed from `lightbox-dialog`.

File: src/less/toast-dialog/toast-dialog.less (hunk 5 above)
Comment (LuLaValva) [reply src/less/toast-dialog/toast-dialog.less:70]: If this isn't included the button compresses vertically and the icon shrinks to a small portion of its size. This is needed.

**Comment by ArtBlue** [comment]
@LuLaValva , let's target `17.5.0` and can you run a Percy build isolated to these dialogs?

**Review by agliga — APPROVED** [review]
Updated to resolve my comments
I wont merge until someone else reviews.

Three changes done:
* Added `align-self: flex-start` to fake-links in header
* Changed all `align-self: start` to be `align-self: flex-start`. I remember there is an issue with start that its not a properly supported property
* Added `overflow-wrap: anywhere`. This prevents really long words for causing a horizontal overflow and rather wraps to the next line. 
### With `overflow-wrap: anywhere`
Screenshot 2024-05-17 at 10 15 50 AM

### Without
Screenshot 2024-05-17 at 10 16 46 AM

**Review by ArtBlue — APPROVED** [review]
Looks perfect now! Thanks!
